### ISSUE TITLE
Test scripts for the layers

### ISSUE THREAD

**Issue by daigo0927** [daigo0927/tf-simple-metric-learning#6]
Implemented layers should be tested compared with some libraries, expecting https://github.com/KevinMusgrave/pytorch-metric-learning or https://github.com/scikit-learn-contrib/metric-learn
